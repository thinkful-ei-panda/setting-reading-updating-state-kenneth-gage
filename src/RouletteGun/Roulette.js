import React, { Component } from 'react' ;


class Roulette extends React.Component {

    static defaultProps = {
        bulletInChamber : 8
    };


       state = {
        chamber : null,
        spinningTheChamber : false,
    } 
  
    

    componentWillUnmount(){
        clearTimeout(this.timeout)
    }



    handleClick(){
        

        this.setState({
            spinningTheChamber : true
        })
        
        this.timeout = setTimeout(()=>{
            this.setState({ chamber :  Math.ceil(Math.random() * 8)  })
            this.setState ({spinningTheChamber : false})

        },2000)
      }

    renderDisplay() {
        console.log(this.props)
        const {chamber, spinningTheChamber} = this.state
        const { bulletInChamber } = this.props
            return spinningTheChamber ? 'spinning the camber '  
            : bulletInChamber === chamber ? 'Bang ' 
            :'your safe' ; 
        
    }

          
         
    


    render(){
        return(
            <div>
                <p>{this.renderDisplay()}</p>
                <button onClick={() => this.handleClick()}>Pull the trigger!</button>
            </div>
        )
    }


}


export default Roulette