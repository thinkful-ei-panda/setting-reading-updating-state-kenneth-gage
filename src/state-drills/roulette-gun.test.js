import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import RouletteGun from './roulette-gun';

describe ( `RouletteGun Component`, () => {
	
	it ( 'renders without crashing', () => {
	  
		const div = document.createElement ( 'div' );
		
		ReactDOM.render ( <RouletteGun />, div );
		
		ReactDOM.unmountComponentAtNode ( div );

	})

	it ( 'renders the UI as expected', () => {

	  expect(

		renderer.create ( <RouletteGun /> ).toJSON ()

	  ).toMatchSnapshot ()

	})

})
