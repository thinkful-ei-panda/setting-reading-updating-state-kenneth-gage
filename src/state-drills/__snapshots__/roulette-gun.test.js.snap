// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouletteGun Component renders the UI as expected 1`] = `
<div>
  <p />
  <button>
    Pull the trigger
  </button>
</div>
`;
